### PR TITLE
[WOE] Implement Imodane, the Pyrohammer

### PR DESCRIPTION
**Pull request by Susucre**
I'll add a few unit tests, but it is working well.

### CODE REVIEW

**Comment by Susucre** [comment]
[[Imodane]]

File: Mage.Sets/src/mage/cards/v/VesuvanDuplimancy.java [hunk 1]
```diff
@@ -29,8 +31,14 @@ public final class VesuvanDuplimancy extends CardImpl {
...
 
+    private static final FilterPermanent subfilter = new FilterControlledPermanent("artifact or creature you control");
+
     static {
-        filter.add(VesuvanDuplimancyPredicate.instance);
+        subfilter.add(Predicates.or(
+                CardType.ARTIFACT.getPredicate(),
+                CardType.CREATURE.getPredicate()
+        ));
+        filter.add(new HasOnlySingleTargetPermanentPredicate(StaticFilters.FILTER_PERMANENT_ARTIFACT_OR_CREATURE));
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/v/VesuvanDuplimancy.java:41]: did you mean to use subfilter here?

File: Mage.Tests/src/test/java/org/mage/test/cards/single/woe/ImodaneThePyrohammerTest.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        // 2 damage to each creature
+        addCard(Zone.HAND, playerA, "Pyroclasm", 1);
+        addCard(Zone.BATTLEFIELD, playerA, "Mountain", 2);
+
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Pyroclasm");
+
+        setStopAt(1, PhaseStep.BEGIN_COMBAT);
+        execute();
+
+        assertLife(playerB, 20); // Only damage on the bears should be accounted for.
```
Comment (xenohedron) [inline Mage.Tests/src/test/java/org/mage/test/cards/single/woe/ImodaneThePyrohammerTest.java:89]: comment shouldn't be copied here

File: Mage.Sets/src/mage/cards/v/VesuvanDuplimancy.java (hunk 1 above)
Comment (Susucre) [reply Mage.Sets/src/mage/cards/v/VesuvanDuplimancy.java:41]: Wait no I don't think I made a mistake there.
filter is the "a spell that targets only a single artifact or creature you control", while subfilter is "artifact or creature you control"

**Review by xenohedron — APPROVED** [review]

**Pull request merged**